File: c ds/linked_list/test2.c
#include<stdio.h>
#include<malloc.h>

struct node{	
	int data;
	struct node *next;
};

struct node *head;

void insert(int x)
{
	struct node *temp;
	temp=(struct node*)malloc(sizeof(struct node));

   temp->data=x;
   temp->next=NULL;
   
   if(head!=NULL)
   {
	temp->next=head;
   }
   head=temp;
}



void disp(struct node *p)
{
	
	while(p!=NULL){
		printf("\n%d",p->data);
		p=p->next;
	}

}

int main()
{
	head=NULL;
	
	int n;
	int x; 
	int k;
	
	printf("how many no u want to add");
	scanf("%d",&n);
	
	for(x=0;x<n;x++)
	{
		printf("\n Enter number");
		scanf("%d",&k);
		insert(k);
	}
	
	disp(head);
}

File: ds1/src/main/java/example1/Test2.java
package example1;

public class Test2 {

    static class Node {
        int data;
        Node next;

        Node(int k) {
            data = k;
        }
    }

    public static void main(String[] args) {
        Node start = new Node(101);
        Node node = start;
        node.next = new Node(102);
        node.next.next = new Node(103);
        node.next.next.next = new Node(104);

        while (start != null) {
            System.out.println(start.data);
            start=start.next;
        }

    }
}
